### PR TITLE
Disable social Connect/Disconnect buttons when user data is refreshing

### PR DESCRIPTION
**Pull request by Tug**
This PR update the social login settings page (at `/me/security/social-login`) to handle disabled button state and errors properly.

### Testing Instructions
- Boot this branch
- Disable 3rd party cookies on your browser ( `chrome://settings/content/cookies` )
- Go to /log-in and check that you see a popover element when you hover the disabled Google Connect button
- Login, go to `/me/security/social-login` and check that the same popover is displayed in this case
- Activate 3rd party cookies
- Sandbox and edit the `/me/social-login/connect` and `/me/social-login/disconnect` endpoints to return an error in all cases
- Check that the error is displayed properly

### Reviews
- [x] Code
- [x] Product

### CODE REVIEW

File: client/state/login/reducer.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export const redirectTo = createReducer( null, {
 	[ SOCIAL_LOGIN_REQUEST_SUCCESS ]: ( state, { data } ) => get( data, 'redirect_to', null ),
 	[ SOCIAL_LOGIN_REQUEST_FAILURE ]: () => null,
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST ]: () => null,
-	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_FAILURE ]: null,
+	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_FAILURE ]: () => null,
```
Comment (Tug) [inline client/state/login/reducer.js]: That's actually quite an important bug fix, it currently makes any action typed `SOCIAL_CONNECT_ACCOUNT_REQUEST_FAILURE` fail to propagate

File: client/components/button/style.scss [hunk 2]
```diff
@@ -122,7 +122,8 @@ button {
 		color: $white;
 	}
 	&[disabled],
-	&:disabled {
+	&:disabled,
+	&.disabled {
 		color: lighten( $gray, 30% );
 		background: $white;
 		border-color: lighten( $gray, 30% );
```
Comment (jsnmoon) [inline client/components/button/style.scss]: What do you think about adding the following styling to the top of this file?
```scss
.button {
    transition: color linear 0.1s, background-color linear 0.1s, border-color linear 0.1s
}
```

File: client/components/social-buttons/google.js [hunk 3]
```diff
@@ -154,34 +154,40 @@ class GoogleLoginButton extends Component {
...
 		const { children } = this.props;
+		let customButton = null;
+
 		if ( children ) {
 			const childProps = {
+				onClick: this.handleClick,
 				onMouseOver: this.showError,
 				onMouseOut: this.hideError,
-				onClick: this.handleClick,
+				className: classNames( { disabled: isDisabled } ),
```
Comment (jsnmoon) [inline client/components/social-buttons/google.js]: Minor: If we're alphabetizing the key names, might as well shoot className to the top 😛

**Comment by yurynix** [comment]
Works as described 👍

File: client/me/social-login/index.jsx [hunk 4]
```diff
@@ -48,11 +53,17 @@ class SocialLogin extends Component {
 		debug( this.constructor.displayName + ' React component is unmounting.' );
 	}
 
+	refreshUser() {
+		user.fetch();
+		this.setState( { fetchingUser: true } );
```
Comment (yurynix) [inline client/me/social-login/index.jsx]: Should we check `if ( ! this.state.fetching ) { ... }` ?

Comment (Tug) [reply client/me/social-login/index.jsx]: I don't think it's necessary. If we're fetching, then the button is disabled and we can't trigger another refresh anyway.
There's however the case where the fetch request fails which is not handled here but that would mean changing some code un lib/user and I'm already investigating this in https://github.com/Automattic/wp-calypso/pull/17772

File: client/components/button/style.scss (hunk 2 above)
Comment (Tug) [reply client/components/button/style.scss]: That's quite a big change that will impact all buttons across calypso. Better to open a new PR for that if you want to suggest it I think

**Comment by scruffian** [comment]
LGTM :+1:

**Pull request merged**